### PR TITLE
Remove gradle dependency constraints, update google BOM version

### PR DESCRIPTION
**Pull request by pshapiro4broad**
Instead of updating constraint for `org.json` in https://github.com/DataBiosphere/terra-common-lib/pull/170 this PR removes the `constraint` section entirely.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -66,7 +66,7 @@ dependencies {
         implementation group: 'com.google.guava', name: 'guava', version: '33.0.0-jre'
```
Comment (okotsopoulos) [inline build.gradle]: Do you know if this constraint is still needed too?  I have a feeling it isn't.

Comment (pshapiro4broad) [reply build.gradle]: You're probably right, let's see

**Review by rtitle — APPROVED** [review]
🙌

**Review by okotsopoulos — APPROVED** [review]
Nice!

**Comment by pshapiro4broad** [comment]
Running a manual scan passed on the branch https://github.com/broadinstitute/dsp-appsec-sourceclear-github-actions/actions/runs/8527995941/job/23360651062

**Pull request merged**